charts/minikube-values.yaml

```diff
@@ -65,14 +53,26 @@ notebooks:
       baseUrl: '/jupyterhub/'
       db:
         type: postgres
-        url: postgres+psycopg2://jupyterhub:d6f59c5df645abe2fd1a0572b6bd5a8d@renku-postgresql:5432/jupyterhub
+        url: postgres+psycopg2://jupyterhub@renku-postgresql:5432/jupyterhub
       services:
         notebooks:
           apiToken: notebookstoken
         gateway:
           admin: true
           oauth_client_id: &gatewayClient gateway
           apiToken: &gatewaySecret 7ed9992425bbdcf03f323310205b2fea34bd9e1a255346f99066800d9440d6f7
+      extraEnv:
+        - name: GITLAB_URL
+          value: https://dev.renku.ch/gitlab
```

Comment:
why is this here? this is wrong, we should delete this extraEnv.
The correct value for minikube is set as part of the upgrade command (see README):
`--set notebooks.jupyterhub.hub.extraEnv.GITLAB_URL=http://$(minikube ip)/gitlab \`

Reply:
Basically:
- To not have the postgres password in the connection string/url, we need to set `PGPASSWORD` in extraEnv
- In order to be able to use a secret value there, we need to set `valueFrom` with a `secretKeyRef`
- `valueFrom` is only available when using `map`representation (`name: <key> value: <value>` instead of `<key>: <value>`) for `extraEnv`, not `list` representation.
- Helm can't merge `map` representations, as it can with `list`, so the highest level that `extraEnv` is defined at HAS to include all the definition of the lower levels, otherwise they'd get lost (overwritten). Which sucks, but is a limitation of helm as far as I can tell.

I don't know enough about our minikube deployment to know the correct values, but we'd need to set it to `http://$(minikube ip)/gitlab` in the `values.yaml` for it to work or use `--set notebooks.jupyterhub.hub.extraEnv[0].value=http://$(minikube ip)/gitlab` on the commandline to modify the single value. But we still need all the `extraEnv`values in every `values.yaml`,  which is a bit annoying.

Reply:
Thanks @pameladelgado for spotting this. Unfortunately, `http://$(minikube ip)/gitlab` in a values file will not work, so `--set notebooks.jupyterhub.hub.extraEnv[0].value=http://$(minikube ip)/gitlab` is the way to go. I adapted the deployment command accordingly.